### PR TITLE
Invert constraints

### LINKED ISSUE #1085: Invert constraints
I can imagine situations where it would be useful to be able to invert constraints so that they do the opposite of what they are designed to do.

```python
from opsdroid.skill import Skill
from opsdroid.matchers import match_regex
from opsdroid.constraints import constrain_users

class MySkill(Skill):

    @match_regex(r'hi')
    @constrain_users(['alice', 'bob'], invert=True)
    async def hello(self, message):
        """Says 'Hey' to anyone EXCEPT 'alice' and 'bob'."""
        await message.respond('Hey')
```

### PR DESCRIPTION
**Pull request by vVv-AA**
Add invert option to constraints.

# Description

Adds invert option for existing constraints.
Adds a negate decorator.
Adds test for inverted constraints.

Fixes #1085

## Status
**READY**

## Type of change

_Please delete options that are not relevant._

- New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Ran existing test suite as per documentation.

- tox

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation (if applicable)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by vVv-AA** [comment]
@jacobtomlinson  

codecov/project Status check seems to be stuck. Can you help resolve this. There are no more details available for this,

File: opsdroid/constraints.py [hunk 1]
```diff
@@ -7,12 +7,13 @@
 import logging
 
 from opsdroid.helper import add_skill_attributes
+from opsdroid.helper import negate
```
Comment (jacobtomlinson) [inline opsdroid/constraints.py]: You could wrap this up into the line above

```
from opsdroid.helper import add_skill_attributes, negate
```

File: opsdroid/constraints.py [hunk 2]
```diff
@@ -23,13 +24,16 @@ def constraint_callback(message, rooms=rooms):
             return message.target in rooms
 
         func = add_skill_attributes(func)
-        func.constraints.append(constraint_callback)
+        if not invert:
```
Comment (jacobtomlinson) [inline opsdroid/constraints.py]: I would prefer if this was not a negative. You could always reassign the function using your negate decorator. For example:

```
if invert:
    constraint_callback = negate(constraint_callback)
```

Then append the contraint callback as before.

Same for all other instances of this pattern.

**Comment by jacobtomlinson** [comment]
Sorry that your change didn't make it in. Hacktoberfest is a competitive time for contributions.

Your PR was awesome though, keep up the good work. I would still be more than happy to send you [opsdroid stickers](https://medium.com/opsdroid/contributor-sticker-packs-738058ceda59)! Fill in the [claim form](https://forms.gle/iQZV9DjkaUiSaDXK6) to get yours!

**Pull request closed without merging**